### PR TITLE
[Docs] Update README/BUILDING for Release

### PR DESCRIPTION
**Pull request by elliotcmorris**
Minimally update README/BUILDING docs to be appropriate for releasing wheels.
This is not the full documentation update, just a quick edit to make the docs make sense.

co-authored by @foundrytom

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -35,7 +75,6 @@ the presence of CMake build targets as exported by these packages.
 
 > Warning: if attempting to build on CentOS 7 using the ConanCenter
 > Python 3.7 package you will likely hit an [issue installing pkgconf](https://github.com/conan-io/conan-center-index/issues/8541).
-
```
[the remark this answers is not in the document]
Comment (elliotcmorris) [reply BUILDING.md]: That's what you get for listening to linters, [9a06da7](https://github.com/OpenAssetIO/OpenAssetIO/pull/705/commits/9a06da70175a56603ca75549eaa20e24864ec3eb)

File: README.md [hunk 2]
```diff
@@ -143,61 +143,25 @@ which appears to overlap with a subset of `OpenAssetIO`s concerns.
...
-are also supported
-
-```shell
-pip install --editable .
+```bash
+python -m pip install openassetio
 ```
 
-#### Quick start: Using the ASWF Docker image
+### Build from source (Using the ASWF Docker image)
```
Comment (feltech) [inline README.md:161]: [consider][minor]
```suggestion
### Build from source (using the ASWF Docker image)
```

File: README.md [hunk 3]
```diff
@@ -143,61 +143,25 @@ which appears to overlap with a subset of `OpenAssetIO`s concerns.
...
+Windows, macOS, and Linux are all supported platforms.
 
-For use in a Python-only environment, the Python package can be
-installed from the root of the repository using
+### Package Installation
 
-```shell
-pip install .
-```
+For pure python projects, OpenAssetIO is available on PyPI, simply run:
```
Comment (feltech) [inline README.md:155]: [consider][minor]
```suggestion
For pure Python projects, OpenAssetIO is available on PyPI, simply run:
```

**Comment by elliotcmorris** [comment]
@feltech Applied both suggestions, rebasing and merging.

**Pull request merged**